lib/finch/http1/pool.ex

```diff
@@ -55,7 +55,23 @@ defmodule Finch.HTTP1.Pool do
     catch
       :exit, data ->
         Telemetry.exception(:queue, start_time, :exit, data, __STACKTRACE__, metadata)
-        exit(data)
+
+        # Provide helpful error messages for known errors
+        case data do
```

Comment:
switched from an `exit` to a `reraise` here to get the output to show up properly. If y'all know of a way to still use `exit` but get this nice error out then I'm happy to switch back to `exit`.